activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -143,6 +149,10 @@ def collector
         end
       end
 
+      def arel_visitor # :nodoc:
+        (Arel::Visitors::VISITORS[@config[:adapter]] || Arel::Visitors::ToSql).new(self)
```

Comment:
This code is derived from https://github.com/rails/arel/blob/v7.0.0/lib/arel/visitors.rb#L33.